docker/docker_client.go

```diff
@@ -419,6 +406,26 @@ func (c *dockerClient) makeRequest(ctx context.Context, method, path string, hea
 // makeRequest should generally be preferred.
 // TODO(runcom): too many arguments here, use a struct
 func (c *dockerClient) makeRequestToResolvedURL(ctx context.Context, method, url string, headers map[string][]string, stream io.Reader, streamLen int64, auth sendAuth, extraScope *authScope) (*http.Response, error) {
+	var (
+		res   *http.Response
+		err   error
+		delay time.Duration
+	)
+	delay = retryAfterDelay
+	for i := 1; i < 3; i++ {
+		res, err = c.makeRequestToResolvedURLHelper(ctx, method, url, headers, stream, streamLen, auth, extraScope)
+		if res != nil && res.StatusCode == http.StatusTooManyRequests {
+			logrus.Errorf("%s: sleeping for %d seconds before next attempt", http.StatusText(res.StatusCode), delay)
+			time.Sleep(delay * time.Second)
```

Comment:
When `i` is 2 I think this will be a useless sleep as we won't retry after it.

Reply:
Good catch!